include/lac/lac_initializer.h

```diff
@@ -59,38 +59,49 @@ class ScopedLACInitializer
   };
 
   /**
-   * Initialize a Trilinos Sparsity Pattern.
+   * Initialize a Trilinos Block Sparse Matrix.
    */
   template<int dim, int spacedim>
-  void operator() (TrilinosWrappers::BlockSparsityPattern &s,
+  void operator() (TrilinosWrappers::BlockSparseMatrix &matrix,
                    const DoFHandler<dim, spacedim> &dh,
                    const ConstraintMatrix &cm,
                    const Table<2,DoFTools::Coupling> &coupling)
   {
-    s.reinit(owned, owned, relevant, comm);
+    TrilinosWrappers::BlockSparsityPattern sp(owned,
+                                              owned,
+                                              relevant,
+                                              comm);
+
     DoFTools::make_sparsity_pattern (dh,
-                                     coupling, s,
+                                     coupling, sp,
                                      cm, false,
                                      Utilities::MPI::this_mpi_process(comm));
-    s.compress();
+    sp.compress();
```

Comment:
@luca-heltai I've noticed that this takes long time.. is it expected (hydrogels, 3x3 blocks)? Moreover, the job  sometimes is **un**balanced between the processors (and/or nodes) inside this function... is this expected?